pkg/manifests/manifests.go

```diff
@@ -1152,19 +1154,32 @@ func (f *Factory) ThanosQuerierRoute() (*routev1.Route, error) {
 }
 
 func (f *Factory) SharingConfig(promHost, amHost, grafanaHost, thanosHost *url.URL) *v1.ConfigMap {
+	data := map[string]string{}
+
+	// Configmap keys need to include "public" to indicate that they are public values.
+	// See https://bugzilla.redhat.com/show_bug.cgi?id=1807100.
+	if promHost != nil {
+		data["prometheusPublicURL"] = promHost.String()
+	}
+
+	if amHost != nil {
+		data["alertmanagerPublicURL"] = amHost.String()
+	}
+
+	if grafanaHost != nil {
+		data["grafanaPublicURL"] = grafanaHost.String()
+	}
+
+	if thanosHost != nil {
+		data["thanosPublicURL"] = thanosHost.String()
+	}
+
```

Comment:
Note: This will be removed in https://github.com/openshift/cluster-monitoring-operator/pull/1223